### PR TITLE
【Task09】Use async await to do the same thing, but don't consider the reject. 

### PR DESCRIPTION
**Pull request by csakyo**
[Task09]
https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#9

[CodeSandBox]
https://codesandbox.io/s/compassionate-yonath-2d3xy

Latest commit: https://github.com/csakyo/homework/pull/19/commits/568714f2dc070a80b7a6cfc60769bad2ac44fd23

[Comment for this Task]
I do not have the confidence about async/await method for asynchronous processing so I would like to know  there is another way to write better? My thinking about processing  is described below comment.

### CODE REVIEW

File: question09/main.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const contentsList = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+                      {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]
+
+
+const promise = new Promise((resolve) => {
+    setTimeout(() =>{
+      resolve(contentsList);
+    }, 3000)
+})
+
```
Comment (csakyo) [inline question09/main.js:20]: ①「contentsList」is resolved in the Promise object.

File: question09/main.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+async function asyncFunc() {
+    const result = await promise;
+    createList(result);
+    loadImg.remove();
+}
+asyncFunc();
+
```
Comment (csakyo) [inline question09/main.js:28]: ②The "await" waits for processing until the result of the specified Promise is returned.
③After the resolved promise is returned, the processes in the asyncFunc function will be executed in order.

File: question09/main.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const aTag = document.createElement('a');
+    aTag.href = to;
+    aTag.innerHTML = text;
+    const imgTag = document.createElement('img');
+    imgTag.src = img;
+    imgTag.alt = alt;
+    fragment.appendChild(liTag).appendChild(aTag).appendChild(imgTag);
+    }
+    ul.appendChild(fragment);
+}
```
Comment (csakyo) [inline question09/main.js:46]: (The functions here are the same as before.)

File: question09/main.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+asyncFunc();
+
+function createList(listElement){
+    const length = listElement.length;
+    const ul = document.getElementById('target');
+    const fragment = document.createDocumentFragment();
+
+    for (let i = 0; i < length; i++){
+    const {to, img, alt, text} = listElement[i];
```
Comment (suefrontend) [inline question09/main.js:35]: Sorry this is not a review, but I think this destructuring part of code is so elegant.

Comment (csakyo) [reply question09/main.js:35]: This part was reviewed before from you! That's why code here is elegant:) Thank you so much for your comment!

File: question09/main.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+// loading img 
+const wrapper = document.getElementById('wrapper');
+const loadImg = document.createElement('img');
+loadImg.src = "loading-circle.gif";
+wrapper.appendChild(loadImg);
```
Comment (kenmori) [inline question09/main.js]: create function and insert `asyncFunc`.

Comment (csakyo) [reply question09/main.js]: I made the function for showing Loading img and inserted to async function. Please confirm.

Latest commit : https://github.com/csakyo/homework/pull/19/commits/1d38e5e3ac222cc1073c6c89a38d3d015f77a2ef

**Pull request merged**